docs/migrations/migrate_renderer.md

```diff
@@ -0,0 +1,228 @@
+- `certificate.additionalData.endDate` [at line 24](https://github.com/OpenCerts/legacy-templates/blob/master/src/components/CertificateTemplates/tlds/sg/gov/tech/2018-Geekout/certificate.js#L24)
+
+In `src/templates/samples/customTemplateSample.ts`, we can update the type (interface `CustomTemplateCertificate`) and the example (variable `customTemplateCertificate`) accordingly:
+
+```typescript
+import { v2 } from "@govtechsg/decentralized-renderer-react-components";
+
+export interface CustomTemplateCertificate extends v2.OpenAttestationDocument {
+  recipient: {
+    name: string;
+  };
+  additionalData: {
+    startDate: string;
+    endDate: string;
+  };
+}
+
+export const customTemplateCertificate: CustomTemplateCertificate = {
+  issuers: [
+    {
+      name: "institute of blockchain"
+    }
+  ],
+  $template: {
+    name: "custom",
+    type: v2.TemplateType.EmbeddedRenderer,
+    url: "http://localhost:3000"
+  },
+  recipient: {
+    name: "John Doe"
+  },
+  additionalData: {
+    startDate: "2019-06-10T00:00:00+08:00",
+    endDate: "2019-06-10T00:00:00+08:00"
+  }
+};
+```
+
+## Creating the template
+
+1. Replace the content of the React component in `src/templates/customTemplate/customTemplate.tsx` by the content of the [2018 Geekout](https://github.com/OpenCerts/legacy-templates/blob/master/src/components/CertificateTemplates/tlds/sg/gov/tech/2018-Geekout/certificate.js). You will see some errors that we will fix in the following steps.
+1. In `src/templates/customTemplate/customTemplate.tsx`:
+
+   1. Update the component declaration
+
+   ```tsx
+   const Template = ({ certificate }) => {
+   ```
+
+   by
+
+   ```tsx
+   export const CustomTemplate: FunctionComponent<TemplateProps<CustomTemplateCertificate>> = ({ document }) => {
+   ```
+
+   This change will help us to make sure our template is correctly typed.
+
+   > The component name does not matter much. However if we change the name `CustomTemplate` we will need to change the whole configuration. To make it simple we will stick to the name that was available in the renderer when we initialized it.
+
+   1. Remove the first line `import PropTypes from "prop-types";` which is useless because of Typescript.
+   1. Remove the 4 last lines which are useless because we don't need prop-types anymore, and the component has already been exported at a previous step.
+
+   ```jsx
+   Template.propTypes = {
+     certificate: PropTypes.object.isRequired
+   };
+   export default Template;
+   ```
+
+   1. Add the following imports
+
+   ```tsx
+   import React, { FunctionComponent } from "react";
+   import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";
+   import { CustomTemplateCertificate } from "../samples";
+   ```
+
+   `TemplateProps`, `CustomTemplateCertificate` and `FunctionComponent` are needed for correctly typing the template.
+
+   1. Add the following imports
```

Comment:
Extra " 1. Add the following imports" which is in line 96

Reply:
Done